src/main/interface/settings.h

```diff
@@ -178,10 +178,11 @@ typedef struct cliArrayLengthConfig_s {
 } cliArrayLengthConfig_t;
 
 typedef union {
-    cliLookupTableConfig_t lookup;
-    cliMinMaxConfig_t minmax;
-    cliArrayLengthConfig_t array;
-    uint8_t bitpos;
+    cliLookupTableConfig_t lookup;  // used for MODE_LOOKUP excl. VAR_UINT32
+    cliMinMaxConfig_t minmax;       // used for MODE_DIRECT
+    cliArrayLengthConfig_t array;   // used for MODE_ARRAY
+    uint8_t bitpos;                 // used for MODE_BITSET
+    uint8_t bitcount;               // used for MODE_DIRECT with VAR_UINT32
```

Comment:
Personally I don't like the `bitcount` implementation. 
There is enough space for `uint32_t u32_max`, it will be cleaner.

And VAR_UINT32 value limit is not necessary now, it may be implemented when it becomes useful

Reply:
Cool, I was torn between bitcount, u32_minmax and u32_max. I like u32_max better too. (It will become useful shortly as I am already working on #5748).